python/cudf/cudf/core/scalar.py

```diff
@@ -353,14 +353,17 @@ def _binaryop(self, other, op: str):
 
     def _dispatch_scalar_binop(self, other, op):
         if isinstance(other, Scalar):
-            other = other.value
+            rhs = other.value
+        else:
+            rhs = other
+        lhs = self.value
```

Comment:
I think we'll always need to come up to the host do do binaryops between scalars because there's no device side binaryop for the underlying device scalar to target in the first place

Reply:
Ah, ok.